From 349e1d571ade97b4cdcd7c5e6c75c886e3cff168 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Mon, 13 Apr 2020 17:57:30 +0300
Subject: [PATCH 138/186] Hide pill in full gestural nav mode [1/2]

  * from
  https://github.com/crdroidandroid/android_frameworks_base/commit/7034894d0dfc6d18e215a66dd76613c4691b4cde

	modified:   core/java/android/provider/Settings.java
	modified:   packages/overlays/Android.mk
	new file:   packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/Android.mk
	new file:   packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/AndroidManifest.xml
	new file:   packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/res/values/config.xml
	new file:   packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/res/values/dimens.xml
	new file:   packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/res/values/strings.xml
	new file:   packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/Android.mk
	new file:   packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/AndroidManifest.xml
	new file:   packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/res/values/config.xml
	new file:   packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/res/values/dimens.xml
	new file:   packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/res/values/strings.xml
	new file:   packages/overlays/NavigationBarModeGesturalOverlayNoPill/Android.mk
	new file:   packages/overlays/NavigationBarModeGesturalOverlayNoPill/AndroidManifest.xml
	new file:   packages/overlays/NavigationBarModeGesturalOverlayNoPill/res/values/config.xml
	new file:   packages/overlays/NavigationBarModeGesturalOverlayNoPill/res/values/dimens.xml
	new file:   packages/overlays/NavigationBarModeGesturalOverlayNoPill/res/values/strings.xml
	new file:   packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/Android.mk
	new file:   packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/AndroidManifest.xml
	new file:   packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/res/values/config.xml
	new file:   packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/res/values/dimens.xml
	new file:   packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/res/values/strings.xml
---
 core/java/android/provider/Settings.java      |  3 +
 packages/overlays/Android.mk                  |  7 ++-
 .../Android.mk                                | 30 ++++++++++
 .../AndroidManifest.xml                       | 27 +++++++++
 .../res/values/config.xml                     | 57 +++++++++++++++++++
 .../res/values/dimens.xml                     | 30 ++++++++++
 .../res/values/strings.xml                    | 22 +++++++
 .../Android.mk                                | 30 ++++++++++
 .../AndroidManifest.xml                       | 27 +++++++++
 .../res/values/config.xml                     | 57 +++++++++++++++++++
 .../res/values/dimens.xml                     | 30 ++++++++++
 .../res/values/strings.xml                    | 22 +++++++
 .../Android.mk                                | 30 ++++++++++
 .../AndroidManifest.xml                       | 27 +++++++++
 .../res/values/config.xml                     | 57 +++++++++++++++++++
 .../res/values/dimens.xml                     | 30 ++++++++++
 .../res/values/strings.xml                    | 22 +++++++
 .../Android.mk                                | 30 ++++++++++
 .../AndroidManifest.xml                       | 27 +++++++++
 .../res/values/config.xml                     | 57 +++++++++++++++++++
 .../res/values/dimens.xml                     | 30 ++++++++++
 .../res/values/strings.xml                    | 22 +++++++
 22 files changed, 673 insertions(+), 1 deletion(-)
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/Android.mk
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/AndroidManifest.xml
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/res/values/config.xml
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/res/values/dimens.xml
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/res/values/strings.xml
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/Android.mk
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/AndroidManifest.xml
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/res/values/config.xml
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/res/values/dimens.xml
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/res/values/strings.xml
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayNoPill/Android.mk
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayNoPill/AndroidManifest.xml
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayNoPill/res/values/config.xml
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayNoPill/res/values/dimens.xml
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayNoPill/res/values/strings.xml
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/Android.mk
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/AndroidManifest.xml
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/res/values/config.xml
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/res/values/dimens.xml
 create mode 100644 packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/res/values/strings.xml

diff --git a/core/java/android/provider/Settings.java b/core/java/android/provider/Settings.java
index 0535ddda1aa..6831ff32fa4 100644
--- a/core/java/android/provider/Settings.java
+++ b/core/java/android/provider/Settings.java
@@ -4495,6 +4495,9 @@ public final class Settings {
         /** @hide */
         public static final String BACK_GESTURE_HEIGHT = "back_gesture_height";
 
+        /** @hide */
+        public static final String GESTURE_PILL_TOGGLE = "gesture_pill_toggle";
+
         /** @hide */
         private static final Validator BACK_GESTURE_HEIGHT_VALIDATOR = ANY_INTEGER_VALIDATOR;
 
diff --git a/packages/overlays/Android.mk b/packages/overlays/Android.mk
index d98b068530b..3255ada35be 100644
--- a/packages/overlays/Android.mk
+++ b/packages/overlays/Android.mk
@@ -95,7 +95,12 @@ LOCAL_REQUIRED_MODULES := \
 	NavigationBarModeGesturalOverlay \
 	NavigationBarModeGesturalOverlayNarrowBack \
 	NavigationBarModeGesturalOverlayWideBack \
-	NavigationBarModeGesturalOverlayExtraWideBack
+	NavigationBarModeGesturalOverlayExtraWideBack \
+	NavigationBarModeGesturalOverlayExtraWideBack \
+	NavigationBarModeGesturalOverlayNoPill \
+	NavigationBarModeGesturalOverlayNarrowBackNoPill \
+	NavigationBarModeGesturalOverlayWideBackNoPill \
+	NavigationBarModeGesturalOverlayExtraWideBackNoPill
 
 include $(BUILD_PHONY_PACKAGE)
 include $(CLEAR_VARS)
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/Android.mk b/packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/Android.mk
new file mode 100644
index 00000000000..4ba7a4c4a55
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/Android.mk
@@ -0,0 +1,30 @@
+#
+#  Copyright 2019, The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_RRO_THEME := NavigationBarModeGesturalExtraWideBackNoPill
+
+
+LOCAL_SRC_FILES := $(call all-subdir-java-files)
+
+LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res
+
+LOCAL_PACKAGE_NAME := NavigationBarModeGesturalOverlayExtraWideBackNoPill
+LOCAL_SDK_VERSION := current
+
+include $(BUILD_RRO_PACKAGE)
\ No newline at end of file
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/AndroidManifest.xml b/packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/AndroidManifest.xml
new file mode 100644
index 00000000000..3daa94ba1f0
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/AndroidManifest.xml
@@ -0,0 +1,27 @@
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+        package="com.android.internal.systemui.navbar.gestural_extra_wide_back_nopill"
+        android:versionCode="1"
+        android:versionName="1.0">
+    <overlay android:targetPackage="android"
+        android:category="com.android.internal.navigation_bar_mode"
+        android:priority="1"/>
+
+    <application android:label="@string/navigation_bar_mode_title" android:hasCode="false"/>
+</manifest>
\ No newline at end of file
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/res/values/config.xml b/packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/res/values/config.xml
new file mode 100644
index 00000000000..d5991f367c4
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/res/values/config.xml
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <!-- Controls the navigation bar interaction mode:
+         0: 3 button mode (back, home, overview buttons)
+         1: 2 button mode (back, home buttons + swipe up for overview)
+         2: gestures only for back, home and overview -->
+    <integer name="config_navBarInteractionMode">2</integer>
+
+    <!-- Controls whether the nav bar can move from the bottom to the side in landscape.
+         Only applies if the device display is not square. -->
+    <bool name="config_navBarCanMove">false</bool>
+
+    <!-- Controls whether the navigation bar lets through taps. -->
+    <bool name="config_navBarTapThrough">true</bool>
+
+    <!-- Controls the size of the back gesture inset. -->
+    <dimen name="config_backGestureInset">40dp</dimen>
+
+    <!-- Controls whether the navbar needs a scrim with
+     {@link Window#setEnsuringNavigationBarContrastWhenTransparent}. -->
+    <bool name="config_navBarNeedsScrim">false</bool>
+
+    <!-- Controls the opacity of the navigation bar depending on the visibility of the
+     various workspace stacks.
+     0 - Nav bar is always opaque when either the freeform stack or docked stack is visible.
+     1 - Nav bar is always translucent when the freeform stack is visible, otherwise always
+         opaque.
+     2 - Nav bar is never forced opaque.
+     -->
+    <integer name="config_navBarOpacityMode">2</integer>
+
+    <!-- Controls whether seamless rotation should be allowed even though the navbar can move
+         (which normally prevents seamless rotation). -->
+    <bool name="config_allowSeamlessRotationDespiteNavBarMoving">true</bool>
+
+    <!-- Controls whether the side edge gestures can always trigger the transient nav bar to
+         show. -->
+    <bool name="config_navBarAlwaysShowOnSideEdgeGesture">true</bool>
+
+</resources>
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/res/values/dimens.xml b/packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/res/values/dimens.xml
new file mode 100644
index 00000000000..66d0ea00c91
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/res/values/dimens.xml
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <!-- Height of the bottom navigation / system bar. -->
+    <dimen name="navigation_bar_height">0dp</dimen>
+    <!-- Height of the bottom navigation bar in portrait; often the same as @dimen/navigation_bar_height -->
+    <dimen name="navigation_bar_height_landscape">0dp</dimen>
+    <!-- Width of the navigation bar when it is placed vertically on the screen -->
+    <dimen name="navigation_bar_width">0dp</dimen>
+    <!-- Height of the bottom navigation / system bar. -->
+    <dimen name="navigation_bar_frame_height">0dp</dimen>
+    <!-- The height of the bottom navigation gesture area. -->
+    <dimen name="navigation_bar_gesture_height">32dp</dimen>
+</resources>
\ No newline at end of file
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/res/values/strings.xml b/packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/res/values/strings.xml
new file mode 100644
index 00000000000..5e45bf95693
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayExtraWideBackNoPill/res/values/strings.xml
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+    <!-- Name of overlay [CHAR LIMIT=64] -->
+    <string name="navigation_bar_mode_title" translatable="false">Gestural Navigation Bar</string>
+</resources>
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/Android.mk b/packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/Android.mk
new file mode 100644
index 00000000000..683bec681a0
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/Android.mk
@@ -0,0 +1,30 @@
+#
+#  Copyright 2019, The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_RRO_THEME := NavigationBarModeGesturalNarrowBackNoPill
+
+
+LOCAL_SRC_FILES := $(call all-subdir-java-files)
+
+LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res
+
+LOCAL_PACKAGE_NAME := NavigationBarModeGesturalOverlayNarrowBackNoPill
+LOCAL_SDK_VERSION := current
+
+include $(BUILD_RRO_PACKAGE)
\ No newline at end of file
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/AndroidManifest.xml b/packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/AndroidManifest.xml
new file mode 100644
index 00000000000..87bd5aae265
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/AndroidManifest.xml
@@ -0,0 +1,27 @@
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+        package="com.android.internal.systemui.navbar.gestural_narrow_back_nopill"
+        android:versionCode="1"
+        android:versionName="1.0">
+    <overlay android:targetPackage="android"
+        android:category="com.android.internal.navigation_bar_mode"
+        android:priority="1"/>
+
+    <application android:label="@string/navigation_bar_mode_title" android:hasCode="false"/>
+</manifest>
\ No newline at end of file
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/res/values/config.xml b/packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/res/values/config.xml
new file mode 100644
index 00000000000..ff507ee9f46
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/res/values/config.xml
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <!-- Controls the navigation bar interaction mode:
+         0: 3 button mode (back, home, overview buttons)
+         1: 2 button mode (back, home buttons + swipe up for overview)
+         2: gestures only for back, home and overview -->
+    <integer name="config_navBarInteractionMode">2</integer>
+
+    <!-- Controls whether the nav bar can move from the bottom to the side in landscape.
+         Only applies if the device display is not square. -->
+    <bool name="config_navBarCanMove">false</bool>
+
+    <!-- Controls whether the navigation bar lets through taps. -->
+    <bool name="config_navBarTapThrough">true</bool>
+
+    <!-- Controls the size of the back gesture inset. -->
+    <dimen name="config_backGestureInset">18dp</dimen>
+
+    <!-- Controls whether the navbar needs a scrim with
+     {@link Window#setEnsuringNavigationBarContrastWhenTransparent}. -->
+    <bool name="config_navBarNeedsScrim">false</bool>
+
+    <!-- Controls the opacity of the navigation bar depending on the visibility of the
+     various workspace stacks.
+     0 - Nav bar is always opaque when either the freeform stack or docked stack is visible.
+     1 - Nav bar is always translucent when the freeform stack is visible, otherwise always
+         opaque.
+     2 - Nav bar is never forced opaque.
+     -->
+    <integer name="config_navBarOpacityMode">2</integer>
+
+    <!-- Controls whether seamless rotation should be allowed even though the navbar can move
+         (which normally prevents seamless rotation). -->
+    <bool name="config_allowSeamlessRotationDespiteNavBarMoving">true</bool>
+
+    <!-- Controls whether the side edge gestures can always trigger the transient nav bar to
+         show. -->
+    <bool name="config_navBarAlwaysShowOnSideEdgeGesture">true</bool>
+
+</resources>
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/res/values/dimens.xml b/packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/res/values/dimens.xml
new file mode 100644
index 00000000000..66d0ea00c91
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/res/values/dimens.xml
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <!-- Height of the bottom navigation / system bar. -->
+    <dimen name="navigation_bar_height">0dp</dimen>
+    <!-- Height of the bottom navigation bar in portrait; often the same as @dimen/navigation_bar_height -->
+    <dimen name="navigation_bar_height_landscape">0dp</dimen>
+    <!-- Width of the navigation bar when it is placed vertically on the screen -->
+    <dimen name="navigation_bar_width">0dp</dimen>
+    <!-- Height of the bottom navigation / system bar. -->
+    <dimen name="navigation_bar_frame_height">0dp</dimen>
+    <!-- The height of the bottom navigation gesture area. -->
+    <dimen name="navigation_bar_gesture_height">32dp</dimen>
+</resources>
\ No newline at end of file
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/res/values/strings.xml b/packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/res/values/strings.xml
new file mode 100644
index 00000000000..5e45bf95693
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayNarrowBackNoPill/res/values/strings.xml
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+    <!-- Name of overlay [CHAR LIMIT=64] -->
+    <string name="navigation_bar_mode_title" translatable="false">Gestural Navigation Bar</string>
+</resources>
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayNoPill/Android.mk b/packages/overlays/NavigationBarModeGesturalOverlayNoPill/Android.mk
new file mode 100644
index 00000000000..2b8458246ad
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayNoPill/Android.mk
@@ -0,0 +1,30 @@
+#
+#  Copyright 2018, The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_RRO_THEME := NavigationBarModeGesturalNoPill
+
+
+LOCAL_SRC_FILES := $(call all-subdir-java-files)
+
+LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res
+
+LOCAL_PACKAGE_NAME := NavigationBarModeGesturalOverlayNoPill
+LOCAL_SDK_VERSION := current
+
+include $(BUILD_RRO_PACKAGE)
\ No newline at end of file
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayNoPill/AndroidManifest.xml b/packages/overlays/NavigationBarModeGesturalOverlayNoPill/AndroidManifest.xml
new file mode 100644
index 00000000000..bb66675b445
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayNoPill/AndroidManifest.xml
@@ -0,0 +1,27 @@
+<!--
+/**
+ * Copyright (c) 2018, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+        package="com.android.internal.systemui.navbar.gestural_nopill"
+        android:versionCode="1"
+        android:versionName="1.0">
+    <overlay android:targetPackage="android"
+        android:category="com.android.internal.navigation_bar_mode"
+        android:priority="1"/>
+
+    <application android:label="@string/navigation_bar_mode_title" android:hasCode="false"/>
+</manifest>
\ No newline at end of file
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayNoPill/res/values/config.xml b/packages/overlays/NavigationBarModeGesturalOverlayNoPill/res/values/config.xml
new file mode 100644
index 00000000000..96ed7b4cc35
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayNoPill/res/values/config.xml
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <!-- Controls the navigation bar interaction mode:
+         0: 3 button mode (back, home, overview buttons)
+         1: 2 button mode (back, home buttons + swipe up for overview)
+         2: gestures only for back, home and overview -->
+    <integer name="config_navBarInteractionMode">2</integer>
+
+    <!-- Controls whether the nav bar can move from the bottom to the side in landscape.
+         Only applies if the device display is not square. -->
+    <bool name="config_navBarCanMove">false</bool>
+
+    <!-- Controls whether the navigation bar lets through taps. -->
+    <bool name="config_navBarTapThrough">true</bool>
+
+    <!-- Controls the size of the back gesture inset. -->
+    <dimen name="config_backGestureInset">24dp</dimen>
+
+    <!-- Controls whether the navbar needs a scrim with
+     {@link Window#setEnsuringNavigationBarContrastWhenTransparent}. -->
+    <bool name="config_navBarNeedsScrim">false</bool>
+
+    <!-- Controls the opacity of the navigation bar depending on the visibility of the
+     various workspace stacks.
+     0 - Nav bar is always opaque when either the freeform stack or docked stack is visible.
+     1 - Nav bar is always translucent when the freeform stack is visible, otherwise always
+         opaque.
+     2 - Nav bar is never forced opaque.
+     -->
+    <integer name="config_navBarOpacityMode">2</integer>
+
+    <!-- Controls whether seamless rotation should be allowed even though the navbar can move
+         (which normally prevents seamless rotation). -->
+    <bool name="config_allowSeamlessRotationDespiteNavBarMoving">true</bool>
+
+    <!-- Controls whether the side edge gestures can always trigger the transient nav bar to
+         show. -->
+    <bool name="config_navBarAlwaysShowOnSideEdgeGesture">true</bool>
+
+</resources>
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayNoPill/res/values/dimens.xml b/packages/overlays/NavigationBarModeGesturalOverlayNoPill/res/values/dimens.xml
new file mode 100644
index 00000000000..66d0ea00c91
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayNoPill/res/values/dimens.xml
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <!-- Height of the bottom navigation / system bar. -->
+    <dimen name="navigation_bar_height">0dp</dimen>
+    <!-- Height of the bottom navigation bar in portrait; often the same as @dimen/navigation_bar_height -->
+    <dimen name="navigation_bar_height_landscape">0dp</dimen>
+    <!-- Width of the navigation bar when it is placed vertically on the screen -->
+    <dimen name="navigation_bar_width">0dp</dimen>
+    <!-- Height of the bottom navigation / system bar. -->
+    <dimen name="navigation_bar_frame_height">0dp</dimen>
+    <!-- The height of the bottom navigation gesture area. -->
+    <dimen name="navigation_bar_gesture_height">32dp</dimen>
+</resources>
\ No newline at end of file
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayNoPill/res/values/strings.xml b/packages/overlays/NavigationBarModeGesturalOverlayNoPill/res/values/strings.xml
new file mode 100644
index 00000000000..f4f95ad4922
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayNoPill/res/values/strings.xml
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2018, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+    <!-- Name of overlay [CHAR LIMIT=64] -->
+    <string name="navigation_bar_mode_title" translatable="false">Gestural Navigation Bar</string>
+</resources>
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/Android.mk b/packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/Android.mk
new file mode 100644
index 00000000000..9b6ccf986fa
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/Android.mk
@@ -0,0 +1,30 @@
+#
+#  Copyright 2019, The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_RRO_THEME := NavigationBarModeGesturalWideBackNoPill
+
+
+LOCAL_SRC_FILES := $(call all-subdir-java-files)
+
+LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res
+
+LOCAL_PACKAGE_NAME := NavigationBarModeGesturalOverlayWideBackNoPill
+LOCAL_SDK_VERSION := current
+
+include $(BUILD_RRO_PACKAGE)
\ No newline at end of file
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/AndroidManifest.xml b/packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/AndroidManifest.xml
new file mode 100644
index 00000000000..79685bcf172
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/AndroidManifest.xml
@@ -0,0 +1,27 @@
+<!--
+/**
+ * Copyright (c) 2018, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+        package="com.android.internal.systemui.navbar.gestural_wide_back_nopill"
+        android:versionCode="1"
+        android:versionName="1.0">
+    <overlay android:targetPackage="android"
+        android:category="com.android.internal.navigation_bar_mode"
+        android:priority="1"/>
+
+    <application android:label="@string/navigation_bar_mode_title" android:hasCode="false"/>
+</manifest>
\ No newline at end of file
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/res/values/config.xml b/packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/res/values/config.xml
new file mode 100644
index 00000000000..378756a50f1
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/res/values/config.xml
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <!-- Controls the navigation bar interaction mode:
+         0: 3 button mode (back, home, overview buttons)
+         1: 2 button mode (back, home buttons + swipe up for overview)
+         2: gestures only for back, home and overview -->
+    <integer name="config_navBarInteractionMode">2</integer>
+
+    <!-- Controls whether the nav bar can move from the bottom to the side in landscape.
+         Only applies if the device display is not square. -->
+    <bool name="config_navBarCanMove">false</bool>
+
+    <!-- Controls whether the navigation bar lets through taps. -->
+    <bool name="config_navBarTapThrough">true</bool>
+
+    <!-- Controls the size of the back gesture inset. -->
+    <dimen name="config_backGestureInset">32dp</dimen>
+
+    <!-- Controls whether the navbar needs a scrim with
+     {@link Window#setEnsuringNavigationBarContrastWhenTransparent}. -->
+    <bool name="config_navBarNeedsScrim">false</bool>
+
+    <!-- Controls the opacity of the navigation bar depending on the visibility of the
+     various workspace stacks.
+     0 - Nav bar is always opaque when either the freeform stack or docked stack is visible.
+     1 - Nav bar is always translucent when the freeform stack is visible, otherwise always
+         opaque.
+     2 - Nav bar is never forced opaque.
+     -->
+    <integer name="config_navBarOpacityMode">2</integer>
+
+    <!-- Controls whether seamless rotation should be allowed even though the navbar can move
+         (which normally prevents seamless rotation). -->
+    <bool name="config_allowSeamlessRotationDespiteNavBarMoving">true</bool>
+
+    <!-- Controls whether the side edge gestures can always trigger the transient nav bar to
+         show. -->
+    <bool name="config_navBarAlwaysShowOnSideEdgeGesture">true</bool>
+
+</resources>
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/res/values/dimens.xml b/packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/res/values/dimens.xml
new file mode 100644
index 00000000000..66d0ea00c91
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/res/values/dimens.xml
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <!-- Height of the bottom navigation / system bar. -->
+    <dimen name="navigation_bar_height">0dp</dimen>
+    <!-- Height of the bottom navigation bar in portrait; often the same as @dimen/navigation_bar_height -->
+    <dimen name="navigation_bar_height_landscape">0dp</dimen>
+    <!-- Width of the navigation bar when it is placed vertically on the screen -->
+    <dimen name="navigation_bar_width">0dp</dimen>
+    <!-- Height of the bottom navigation / system bar. -->
+    <dimen name="navigation_bar_frame_height">0dp</dimen>
+    <!-- The height of the bottom navigation gesture area. -->
+    <dimen name="navigation_bar_gesture_height">32dp</dimen>
+</resources>
\ No newline at end of file
diff --git a/packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/res/values/strings.xml b/packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/res/values/strings.xml
new file mode 100644
index 00000000000..5e45bf95693
--- /dev/null
+++ b/packages/overlays/NavigationBarModeGesturalOverlayWideBackNoPill/res/values/strings.xml
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+    <!-- Name of overlay [CHAR LIMIT=64] -->
+    <string name="navigation_bar_mode_title" translatable="false">Gestural Navigation Bar</string>
+</resources>
-- 
2.17.1

